ablateLibrary/domain/subDomain.cpp

```diff
@@ -137,7 +166,45 @@ PetscObject ablate::domain::SubDomain::GetPetscFieldObject(const Field& field) {
 
 void ablate::domain::SubDomain::ProjectFieldFunctions(const std::vector<std::shared_ptr<mathFunctions::FieldFunction>>& initialization, Vec globVec, PetscReal time) {
     PetscInt numberFields;
-    auto dm = GetDM();
+    DM dm;
+
+    VecGetDM(globVec, &dm) >> checkError;
+    if (dm != GetDM()) {
+        throw std::invalid_argument("The Vector passed in to ablate::domain::SubDomain::ProjectFieldFunctions must be a global vector from the DM.");
+    }
+    DMGetNumFields(dm, &numberFields) >> checkError;
+
+    // size up the update and context functions
+    std::vector<mathFunctions::PetscFunction> fieldFunctions(numberFields, NULL);
+    std::vector<void*> fieldContexts(numberFields, NULL);
+
+    for (auto fieldInitialization : initialization) {
+        auto fieldId = GetSolutionField(fieldInitialization->GetName());
+
+        fieldContexts[fieldId.id] = fieldInitialization->GetSolutionField().GetContext();
+        fieldFunctions[fieldId.id] = fieldInitialization->GetSolutionField().GetPetscFunction();
+    }
+
+    if (label == nullptr) {
+        DMProjectFunction(dm, time, &fieldFunctions[0], &fieldContexts[0], INSERT_VALUES, globVec) >> checkError;
+
+    } else {
+        DMProjectFunctionLabel(dm, time, label, region->GetValues().size(), &region->GetValues()[0], -1, NULL, &fieldFunctions[0], &fieldContexts[0], INSERT_VALUES, globVec);
+    }
+}
+
+void ablate::domain::SubDomain::ProjectFieldFunctionsToSubDM(const std::vector<std::shared_ptr<mathFunctions::FieldFunction>>& initialization, Vec globVec, PetscReal time) {
```

Comment:
Note that the `dm` input to Project can have different fields than the DM attached to the input vector, so you can compress out fields with the single call if you want.

Reply:
Thanks!  Does that mean that the fieldFunctions should be sized/correspond to vec and not the dm?

Reply:
I do not completely understand the question, but we could go over it at the meeting.